### ISSUE TITLE
Unit tests failing under Python 3.7

### ISSUE THREAD

**Issue by FedericoCeratto** [bottlepy/bottle#1190]
Hello and thanks for developing Bottle. The tests in Bottle 0.12.18 are failing under Python 3.7.5, both with tox (configured with a py37 testenv) and without:

```bash
tox 2>&1 | grep -e ^FA -e ^Assert
FAIL: test_delete_cookie (test_environ.TestResponse)
AssertionError: False is not true
FAIL: test_expires_header (test_environ.TestResponse)
AssertionError: 0 != 3600
FAIL: test_asctime (test_sendfile.TestDateParser)
AssertionError: 1576965590 != 1576961990
FAIL: test_rfc1123 (test_sendfile.TestDateParser)
AssertionError: 1576965590 != 1576961990
FAIL: test_rfc850 (test_sendfile.TestDateParser)
AssertionError: 1576965590 != 1576961990
FAIL: test_ims (test_sendfile.TestSendFile)
AssertionError: 1484330560 != 1484326960.0
FAILED (failures=6)
```

Most of the failures also happen under Python 2.7.17

**Comment by defnull** [comment]
I can only reproduce the delete_cookie error (python 3.5 changed the way empty cookies are rendered). All other tests pass with Python `2.[5-7]` and `3.[2-7]` installed via deadsnakes ppa in an Ubuntu 18.04 docker container. I tried it with and without `-eTZ=Europe/Berlin` to ensure there are no hidden timezone-related errors.

**Comment by FedericoCeratto** [comment]
Interesting. I'm seeing failures in test_delete_cookie but also `test_invalid (test_sendfile.TestSendFile)` on both:

https://salsa.debian.org/debian/python-bottle/-/jobs/482151 and
https://circleci.com/gh/FedericoCeratto/bottle/2

The timezone issues, instead, seem to affect only my workstation.

**Comment by defnull** [comment]
Cookie test is fixed now in release-0.12 branch (will be part of next release, whenever that happens).
The sendfile test fails because you are running tests as root. The test checks what happens if you try to fetch a file with missing read permissions. Root can read any files, so the expected error does not happen.
